src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1512,6 +1512,61 @@ protected virtual void AssignImpl(BoundNode node, BoundExpression value, bool is
                         break;
                     }
 
+                case BoundKind.InlineArrayAccess:
+                    {
+                        var elementAccess = (BoundInlineArrayAccess)node;
+
+                        // PROTOTYPE(InlineArrays): Ensure adequate test coverage
+
+                        switch (elementAccess.Expression.Kind)
+                        {
+                            case BoundKind.Local:
+                                {
+                                    var local = (BoundLocal)elementAccess.Expression;
+                                    if (local.LocalSymbol.RefKind != RefKind.None && !isRef)
+                                    {
+                                        // Writing through the (reference) value of a reference local
+                                        // requires us to read the reference itself.
+                                        if (written) VisitRvalue(local, isKnownToBeAnLvalue: true);
+                                    }
+                                    else
+                                    {
+                                        if (written) NoteWrite(local, null, read);
+                                    }
+                                    break;
+                                }
+
+                            case BoundKind.InlineArrayAccess:
+                                AssignImpl(elementAccess.Expression, null, isRef, written, read);
+                                break;
+
+                            case BoundKind.Parameter:
+                                {
+                                    var paramExpr = (BoundParameter)elementAccess.Expression;
+                                    var param = paramExpr.ParameterSymbol;
+                                    // If we're ref-reassigning an out parameter we're effectively
+                                    // leaving the original
+                                    if (isRef && param.RefKind == RefKind.Out)
+                                    {
+                                        LeaveParameter(param, elementAccess.Expression.Syntax, paramExpr.Syntax.Location);
+                                    }
+
+                                    if (written) NoteWrite(paramExpr, null, read);
+                                    break;
+                                }
+
+                            case BoundKind.ThisReference:
```

Comment:
>ThisReference [](http://example.com/codeflow?start=43&length=13)

Do we need to handle `BaseReference` as well?

Reply:
> Do we need to handle ```BaseReference``` as well?

I do not think so. It looks like this function doesn't track assignements to the base. If you have a specific scenario in mind, feel free to share.